src/common/types/Aliases.ts

```diff
@@ -26,3 +29,18 @@ export function notNull<TValue> (value: TValue | null | undefined): value is TVa
  * This is an intersection of NPM log levels and 'loglevel' library methods.
  */
 export type NpmLogLevel = 'error' | 'warn' | 'info' | 'debug'
+
+export type Web3Provider =
+  | HttpProvider
+  | IpcProvider
+  | WebsocketProvider
+
+/**
+ * The only thing that is guaranteed a Web3 provider or a similar object is a {@link send} method.
+ */
+export interface ISendProvider {
```

Comment:
Done. Actually, `Web3Provider` is already defined in a few places, so to avoid any chance of confusion I named it `Web3ProviderBaseInterface`. Hope this is better 😂